tinyagentos/projects/canvas/unfurl.py

```
@@ -0,0 +1,128 @@
+        "favicon_url": "",
+        "fetched_at": time.time(),
+    }
+
+
+async def fetch_link_metadata(url: str) -> dict:
+    try:
+        status, body = await asyncio.wait_for(_http_get(url), timeout=_TIMEOUT + 0.5)
+    except (asyncio.TimeoutError, TimeoutError):
+        logger.info("canvas unfurl timeout for %s", url)
+        return _fallback(url)
+    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)